src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,97 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_LENGTH = 10;
+    private static final double MULTIPLIER = 1.5;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_LENGTH];
+    }
+
+    public void grow() {
+        int newLength = (int) (elementData.length * MULTIPLIER);
+        T[] elementDataNewLength = (T[]) new Object[newLength];
+        System.arraycopy(elementData, 0, elementDataNewLength, 0, elementData.length);
+        elementData = elementDataNewLength;
+    }
+
+    public void growingCheck() {
+        if (size == elementData.length) {
+            grow();
+        }
+    }
```

Comment:
do not forget to encapsulate the internal functionality of the class by making it private